### PR TITLE
[neox] network

### PR DESCRIPTION
**Pull request by KickSeason**
Change #1703 to `neox`.
### mpt integration (Step 3/4) 
* P2P messsage `StateRoot` and `GetStateRoot` for state root sync
* Request sync state root in `TaskManager`

### CODE REVIEW

File: src/neo/Network/P2P/Payloads/StateRoot.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using System;
+using System.IO;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class StateRoot : ICloneable<StateRoot>, IInventory
```
Comment (shargon) [inline src/neo/Network/P2P/Payloads/StateRoot.cs:13]: If we decide to save it in block header, i think that this wont be required.

Comment (KickSeason) [reply src/neo/Network/P2P/Payloads/StateRoot.cs:13]: Yes. But Erik doesn't agree.

File: src/neo/Ledger/Blockchain.State.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+using Akka.Actor;
+using Neo.Network.P2P.Payloads;
+
+namespace Neo.Ledger
+{
+    public sealed partial class Blockchain : UntypedActor
+    {
+        public StateRoot LatestValidatorsStateRoot => new StateRoot();
```
Comment (Tommo-L) [inline src/neo/Ledger/Blockchain.State.cs:8]: It's not good to split a new `Blockchain.State.cs` file with too little code.

Comment (shargon) [reply src/neo/Ledger/Blockchain.State.cs:8]: Agree

**Pull request merged**